routes/v1/gardens/gardens_router.go

```diff
@@ -101,14 +113,48 @@ func UpdateGardenById(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 }
-func GetGardensByUserId(w http.ResponseWriter, r *http.Request) {
+
+func DeleteGardenByGardenId(w http.ResponseWriter, r *http.Request)  {
+	fmt.Println("Endpoint hit: delete garden by id")
 	vars := mux.Vars(r)
 	utils.EnableCors(&w)
-	paramsUserId := vars["userFireBaseId"]
 
-	userGardens := gardens_controllers.GetGardensByUserId(paramsUserId)
+	paramsGardenId := vars["gardenId"]
 
-	w.Header().Set("Content-Type", "application/json")
-	json.NewEncoder(w).Encode(userGardens)
+	oid, err := primitive.ObjectIDFromHex(paramsGardenId)
 
-}
+	if err != nil {
+		log.Println("Error converting params gardenId to ObjectId:", err)
+		w.Header().Set("Content-Type", "application/json")
+		var iceCreamError errors_models.IceCreamErrors
+		iceCreamError.Error = err.Error()
+		iceCreamError.Info = "Invalid gardenId provided"
+		json.NewEncoder(w).Encode(iceCreamError)
```

Comment:
just curious, `iceCreamError` Using iceCream for naming, is something decided in the backend team? its related to the repo?

Reply:
@daiHash No, I just put it in there for now. I didn't want to name it just Errors because it sounded too generic.